src/System.Linq.Expressions/tests/StackSpillerTests.cs

```diff
@@ -375,82 +373,1312 @@ public static void Spill_LabelGoto()
+                        [1] valuetype [System.Linq.Expressions.Tests]System.Linq.Expressions.Tests.StackSpillerTests+ValueBar&,
+                        [2] int32,
+                        [3] valuetype [System.Linq.Expressions.Tests]System.Linq.Expressions.Tests.StackSpillerTests+ValueBar,
+                        [4] valuetype [System.Linq.Expressions.Tests]System.Linq.Expressions.Tests.StackSpillerTests+ValueBar,
+                        [5] int32
+                      )
+
+                      // t = new ValueBar()
+                      IL_0000: ldloca.s   V_4
+                      IL_0002: initobj    valuetype [System.Linq.Expressions.Tests]System.Linq.Expressions.Tests.StackSpillerTests+ValueBar
+                      IL_0008: ldloc.s    V_4
+                      IL_000a: stloc.0    
+
+                      // V0 = ref t                  // spill reference to t [V0] into V1
+                      IL_000b: ldloca.s   V_0
+                      IL_000d: stloc.1    
+
+                      // V2 = try { 42 } finally { }
+                      .try
+                      {
+                        IL_000e: ldc.i4.s   42
+                        IL_0010: stloc.s    V_5
+                        IL_0012: leave      IL_0018
+                      }
+                      finally
+                      {
+                        IL_0017: endfinally 
+                      }
+                      IL_0018: ldloc.s    V_5
+                      IL_001a: stloc.2    
+
+                      // t.Baz = try { 42 }          // using spilled reference in V0
+                      IL_001b: ldloc.1    
+                      IL_001c: ldloc.2    
+                      IL_001d: stfld      valuetype [System.Linq.Expressions.Tests]System.Linq.Expressions.Tests.StackSpillerTests+ValueBar::Baz
+
+                      // l = t
+                      IL_0022: ldloc.0    
+                      IL_0023: stloc.3    
+
+                      // return l
+                      IL_0024: ldloc.3    
+                      IL_0025: ret        
+                    }",
+                instructions: @"
+                    object lambda_method(object[])
+                    {
+                      .locals 1
+                      .maxstack 5
+                      .maxcontinuation 1
+
+                      IP_0000: InitMutableValue(0)
+                      IP_0001: New System.Linq.Expressions.Tests.StackSpillerTests+ValueBar
+                      IP_0002: Dup()
+                      .try
+                      {
+                        IP_0003: EnterTryFinally[0] -> 6
+                        IP_0004: LoadObject(42)
+                        IP_0005: Goto[1] -> 8
+                      }
+                      finally
+                      {
+                        IP_0006: EnterFinally[0] -> 6
+                        IP_0007: LeaveFinally()
+                      }
+                      IP_0008: StoreField()
```

Comment:
Will review the instruction printing separately later; this should include the field name for sure. It'd likely be a good idea to review all reflection `ToString` usage here because many of them don't quite produce enough information to see what's going on (e.g. the string representation of `MethodInfo` does not include the declaring type).